.github/workflows/main.yml

```diff
@@ -16,6 +16,8 @@ jobs:
 
   Build:
     name: Build
+    permissions:
```

Comment:
the permission needs to be set where the workflow is being called as well, otherwise the steps fail as it won't be allowed to request the permissions